client/src/legacy/TabSet.js

```diff
@@ -36,6 +40,46 @@ $.entwine('ss', function($){
         this._super();
     },
 
+    restoreState: function (overrideState) {
```

Comment:
This functionality is mostly copied directly from `LeftAndMain.js` with some minor changes:
- some of the code has been slightly modernised
- if the tabset hasn't been initialised yet, is is marked to defer restoring the state until `redrawTabs()` is called.